components/Tabs/src/private/MDCTabBarIndicatorView.m

```diff
@@ -0,0 +1,94 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTabBarIndicatorView.h"
+
+#import "MDCTabBarIndicatorAttributes.h"
+
+/** Content view that displays a filled path and supports animation between states. */
+@interface MDCTabBarIndicatorShapeView: UIView
+
+/** The path to display. It will be filled using the view's tintColor. */
+@property(nonatomic, nullable) UIBezierPath *path;
+
+@end
+
+@implementation MDCTabBarIndicatorView {
+  /// View responsible for drawing the indicator's path.
+  MDCTabBarIndicatorShapeView *_shapeView;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    // Fill the indicator with the shape.
+    _shapeView = [[MDCTabBarIndicatorShapeView alloc] initWithFrame:self.bounds];
+    _shapeView.autoresizingMask =
+        UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    [self addSubview:_shapeView];
+  }
+  return self;
+}
+
```

Comment:
If we implement one designated initializer, we likely need all designated initializers.
Do we need to implement initWithCoder: ?
Confirm this works as expected in a sample?

Reply:
We shouldn't need it, but implemented anyway. Only the top-level MDCTabBar should need to support NSCoding, as it directly instantiates all of its children